### PR TITLE
Implement new manual booking links

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Ticket: https://projects.torchbox.com/projects/rca-django-cms-project/tickets/1096

### CODE REVIEW

File: rca/shortcourses/models.py [hunk 1]
```diff
@@ -332,6 +409,15 @@ def clean(self):
             errors["contact_url"].append(
                 "Only one of URL or an Email value is supported here"
             )
+        if (
+            not self.access_planit_course_id
+            and not self.manual_registration_url
+            and self.show_register_link
```
Comment (tombola) [inline rca/shortcourses/models.py:415]: **suggestion** (personal): This _might_ be readable with the positive condition first, and the negation grouped:

```
if self.show_register_link and not (
    self.access_planit_course_id and self.manual_registration_url
):
```

File: rca/shortcourses/tests/test_models.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        )
+
+        response = self.client.get("/short-course/")
+        self.assertContains(response, "Short course title")
+        self.assertNotIn("Register your interest for upcoming dates", response)
+        self.assertContains(response, "Applications are now closed")
+        self.assertNotIn(register_link, response)
+        self.assertEqual(response.render().status_code, 200)
+
+    def test_no_data_at_and_register_link(self):
```
Comment (tombola) [inline rca/shortcourses/tests/test_models.py:89]: **nitpick**: function name does not scan (`_at_`)

File: rca/shortcourses/tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            "register_interest_link"
+        ]
+
+        booking_bar_data = self.short_course_page._format_booking_bar(
+            register_interest_link=register_link, access_planit_data=None
+        )
+
+        self.assertEqual(
+            {
+                "message": "Applications are now closed",
```
Comment (tombola) [inline rca/shortcourses/tests/test_models.py:75]: **nitpick**: This string is referenced in a few places, could it be a constant (for maintenance)

File: rca/shortcourses/tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                "action": "Register your interest for upcoming dates",
```
Comment (tombola) [inline rca/shortcourses/tests/test_models.py:76]: **nitpick**: This string is referenced in a few places, could it be a constant (for maintenance)

**Review by tombola — COMMENTED** [review]
This all generally looks good to me, my eyes started to cross a little reading the tests, but they _look_ sensible.

A few comments.

File: rca/project_styleguide/templates/patterns/molecules/upcoming-dates/upcoming-dates.html [hunk 5]
```diff
@@ -4,33 +4,60 @@
...
+                                <span class="link__label">Register interest</span>
+                                <svg width="11" height="13" class="link__icon" aria-hidden="true"><use xlink:href="#arrow-external"></use></svg>
+                            </a>
+                        {% else %}
+                            <a href="{{ item.booking_link }}" class="link link--tertiary link--link link--external" target="_blank">
+                                <span class="link__label">Book</span>
+                                <svg width="11" height="13" class="link__icon" aria-hidden="true"><use xlink:href="#arrow-external"></use></svg>
+                            </a>
+                        {% endif %}
+                    </div>
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply rca/project_styleguide/templates/patterns/molecules/upcoming-dates/upcoming-dates.html]: Thanks fixed in https://github.com/torchbox/rca-wagtail-2019/pull/484/commits/f41f36983f1cd6e039351c477167d1edc159a5b7

**Pull request merged**